### PR TITLE
moved cloneable to annotation package

### PR DESCRIPTION
**Pull request by lrytz**
requires new starr.

review by @jsuereth

the first commit allows building the new starr: just run the script `tools/new-starr` (i didn't have time to do that in ant right now..)

the second commit is the new starr and removes the temporary files in `src/starr`

### CODE REVIEW

**Comment by lrytz** [comment]
so we have two pull requests now that require a new starr (also #825) - one of them will have to be re-submitted. i can easily re-submit mine once the other gets merged.

File: build.xml [hunk 1]
```diff
@@ -1144,7 +1144,7 @@ QUICK BUILD (QUICK)
     </javac>
     <scalacfork
       destdir="${build-quick.dir}/classes/compiler"
-      compilerpathref="starr.classpath"
+      compilerpathref="locker.classpath"
```
Comment (jsuereth) [inline build.xml:1153]: WHAT!!!?????  This is terrible, and it's all my fault!  THanks much for the fix.

File: tools/new-starr [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+ant -Dscalac.args.optimise="-optimise" locker.done
+cp -R src/starr/* src/
```
Comment (jsuereth) [inline tools/new-starr:4]: Not bad.   I can make this part of the ant targets in the future, so we don't remove the original source _and_ we still stop using this after a new starr....

For now this is 100% better than what we were doing.  Thanks much!

**Comment by jsuereth** [comment]
LGTM where G = great

**Comment by lrytz** [comment]
replacestarr generates a jar without `asm`?

**Comment by jsuereth** [comment]
You probably need my pull request first then.

**Comment by lrytz** [comment]
PLS REBUILD ALL

**Comment by lrytz** [comment]
Hi @jsuereth, could you give a final review for this one (including the source code changes related to `@cloneable` / `extends Cloneable`)?

File: src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala [hunk 3]
```diff
@@ -1368,7 +1368,7 @@ abstract class GenASM extends SubComponent with BytecodeWriters {
         // Additional interface parents based on annotations and other cues
         def newParentForAttr(attr: Symbol): Option[Symbol] = attr match {
           case SerializableAttr => Some(SerializableClass)
-          case CloneableAttr    => Some(JavaCloneableClass)
```
Comment (jsuereth) [inline src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala:1371]: Shouldn't we also support the java convention?

Comment (lrytz) [inline src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala:1371]: `scala.Cloneable` is a subtype of `java.lang.Cloneable`. Both would work, I think it doesn't matter too much in the end: anyway the `@cloneable` annotation is deprecated now.

Comment (jsuereth) [inline src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala:1371]: NVM - I see.  This  adds the cloneable class as a parent.  I was misreading what this method does....

**Comment by jsuereth** [comment]
LGTM - However, you may need to rebase again.  I fixed that build issue for M5 to make sure we had valid MSIL bytecode in m5

**Comment by lrytz** [comment]
it seems we can still merge automatically, since i did the exact same changes as you.

**Comment by adriaanm** [comment]
I tried rebasing this but couldn't figure it out, sorry, Lukas.
Could you please submit a new pull req against 2.10.x? thanks!

**Pull request closed without merging**